Desactivate login option

cm0s opened the pull request:
Here is another pull request :o). It's not urgent!

With this modification you can add a "shib.login=false" to the application.conf. 
When the value is set to false the module doesn't redirect the user to the login url set with the "shib.login.url" property. This capability to desactivate login is necessary when you prefer to set the shibboleth login redirection at the level of the web server and don't want to let the application handle the redirection (it's a bit more secure).

By default the "shib.login" is set to true (so it can be omitted in the application.conf).

If you think this modification can be useful I can also pull a request to add a test and eventually a documentation modification.

And again, thank you for this great plugin!

cm0s committed:
Added shibboleth login activation/desactivation capability

cm0s commented:
This is the corrected pull request. In this one the authenticate action is correctly called, even if the "shib.login" property is set to false.

scott-phillips commented:
cm0s,

I've merged in your changes. I have also updated the documentation for the module for both changes. I created a new term (i think) called restricted sessions to describe the use case where you need shib.login=false. I provided example configuration for both models. Hopefully people won't get too confused. Here at A&M we don't really have any applications that we completely restrict access to, it's almost always the case where anonymous users need access as well. But it is definitely a case that needs to be covered by the module.

One important thing. I changed the default logic for "shib.login" to be false by default. I realize this causes some backwards compatibility problems but it makes sense holistically. I can't imagine too many people are using the module already so I think the needs of the few out weight the needs of the many. It means that if your application is using "ShibRequireSessions ON" that you don't need to provide any additional configuration beyond the attribute mapping. You only need to worry about the initiator configuration if you are going to use it. It just made sense to me.

I have rebuilt a head version of the module and submitted it to the Play repository.

I believe that the module as it exists is ready to cut a next version with your two changes. Are you planning on making any more pull requests?

Scott--

cm0s commented:
Hi Scott,

Thank you for your modifications, I hope it will be useful for other people.
Right now, I don't have any other pull request and I also think you can
release a new module version.

Regards,

Nicolas

On Mon, Jan 23, 2012 at 11:32 PM, Scott Phillips <
reply@reply.github.com

> wrote:
> 
> cm0s,
> 
> I've merged in your changes. I have also updated the documentation for the
> module for both changes. I created a new term (i think) called restricted
> sessions to describe the use case where you need shib.login=false. I
> provided example configuration for both models. Hopefully people won't get
> too confused. Here at A&M we don't really have any applications that we
> completely restrict access to, it's almost always the case where anonymous
> users need access as well. But it is definitely a case that needs to be
> covered by the module.
> 
> One important thing. I changed the default logic for "shib.login" to be
> false by default. I realize this causes some backwards compatibility
> problems but it makes sense holistically. I can't imagine too many people
> are using the module already so I think the needs of the few out weight the
> needs of the many. It means that if your application is using
> "ShibRequireSessions ON" that you don't need to provide any additional
> configuration beyond the attribute mapping. You only need to worry about
> the initiator configuration if you are going to use it. It just made sense
> to me.
> 
> I have rebuilt a head version of the module and submitted it to the Play
> repository.
> 
> I believe that the module as it exists is ready to cut a next version with
> your two changes. Are you planning on making any more pull requests?
> 
> Scott--
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/TAMULib/Shibboleth-play/pull/3#issuecomment-3623365
